### PR TITLE
Added shim that allows us to use m2r components, but recommonmark mainly

### PR DESCRIPTION
**Pull request by dotsdl**
Addresses #598 

This PR fixes the doc build, which uses Sphinx 3.x.  `m2r`, which we used for markdown handling, appears unmaintained and fails when building the docs under this version of Sphinx.

I used [this approach](https://github.com/readthedocs/recommonmark/issues/191#issuecomment-622369992) to use the components we need from `m2r` (`.. mdinclude::` directive), but `recommonmark` for everything else.

I would like a second set of eyes on the built docs to ensure everything is being generated as expected.

Also made some changes to makefile in line with latest sphinx (3.x) config generation. This may not necessarily be desirable; review requested.

- [x] Tag issue being addressed
- [x] Add [tests](https://github.com/openforcefield/openforcefield/tree/master/openforcefield/tests)
- [x] Update docstrings/[documentation](https://github.com/openforcefield/openforcefield/tree/master/docs), if applicable
- [x] Update [changelog](https://github.com/openforcefield/openforcefield/blob/master/docs/releasehistory.rst)

### CODE REVIEW

**Comment by dotsdl** [comment]
[[Toolkit] Docs are broken](https://trello.com/c/rPLZWTVz/22-toolkit-docs-are-broken)

File: docs/conf.py [hunk 1]
```diff
@@ -26,6 +26,9 @@
 
 import openforcefield
 
+from recommonmark.transform import AutoStructify
```
Comment (mattwthompson) [inline docs/conf.py:29]: This was not pulled down for me with `conda env create --name openff-docs --file environment.yml`

**Comment by mattwthompson** [comment]
This builds locally for me and looks good (after grabbing `recommonmark`). I looked over a handful of pages and nothing seemed out of the ordinary - anything in particular to check?

**Comment by j-wags** [comment]
Looking at this now -- I've tagged this branch for builds on RTD, so we should be able to see status here: https://readthedocs.org/projects/open-forcefield-toolkit/builds/

**Comment by dotsdl** [comment]
Just tried switching `m2r_parse_relative_links` to `True` as @j-wags suggested in [here](https://github.com/openforcefield/openforcefield/pull/602#pullrequestreview-415684493). Appears to produce the [same effect](https://open-forcefield-toolkit.readthedocs.io/en/doc-fix/releasehistory.html#id35). :cry:

**Comment by dotsdl** [comment]
Tried [this approach](https://github.com/miyakogi/m2r/issues/51#issuecomment-618285433) locally just now. Doesn't require `recommonmark`, but also doesn't fix the issue with broken API links. Going to push this so our dependencies remain the same, though it doesn't advance our approach much further.

**Comment by j-wags** [comment]
I wonder if it'd be worthwhile to take another go at pinning to an old version of sphinx. If we can force it down to `2.4.4`, then all of our problems are solved (for a while).

**Comment by dotsdl** [comment]
> I wonder if it'd be worthwhile to take another go at pinning to an old version of sphinx. If we can force it down to 2.4.4, then all of our problems are solved (for a while).

Just tried to set `sphinx == 2.4` in conda `docs/environment.yml`, but it looks like [the build](https://readthedocs.org/projects/open-forcefield-toolkit/builds/11100212/) installs sphinx after running this, overriding that specification. Any ideas on other knobs we can turn here?

**Comment by jaimergp** [comment]
Maybe this helps: https://github.com/readthedocs/readthedocs.org/issues/5810

**Comment by j-wags** [comment]
We might be able to force a "conda" style build instead of a "sphinx" style one, and then just manually install the dependencies, including `sphinx=2`, as conda packages.

**Comment by dotsdl** [comment]
> We might be able to force a "conda" style build instead of a "sphinx" style one, and then just manually install the dependencies, including `sphinx=2`, as conda packages.

We are already doing a `conda` style build. I have [added an explicit sphinx pin](https://github.com/openforcefield/openforcefield/pull/602/files#diff-b0278a395ac6448087984d5186dbe855R11) to our `docs/environment.yml`, but this gets overwritten by [the build](https://readthedocs.org/projects/open-forcefield-toolkit/builds/11124310/), which runs the following right after installing our environment:

```
> conda install --yes --quiet --name doc-fix mock pillow sphinx sphinx_rtd_theme
Collecting package metadata (current_repodata.json): ...working... done
Solving environment: ...working... done

## Package Plan ##

  environment location: /home/docs/checkouts/readthedocs.org/user_builds/open-forcefield-toolkit/conda/doc-fix

  added / updated specs:
    - mock
    - pillow
    - sphinx
    - sphinx_rtd_theme


The following packages will be downloaded:

    package                    |            build
    ---------------------------|-----------------
    ca-certificates-2020.1.1   |                0         125 KB
    certifi-2020.4.5.1         |           py37_0         155 KB
    mock-4.0.2                 |             py_0          32 KB
    openssl-1.1.1g             |       h7b6447c_0         2.5 MB
    pillow-7.1.2               |   py37hb39fc2d_0         603 KB
    sphinx-3.0.3               |             py_0         1.1 MB
    sphinx_rtd_theme-0.4.3     |             py_0         4.4 MB
    ------------------------------------------------------------
                                           Total:         9.0 MB

The following NEW packages will be INSTALLED:

  mock               pkgs/main/noarch::mock-4.0.2-py_0
  sphinx_rtd_theme   pkgs/main/noarch::sphinx_rtd_theme-0.4.3-py_0

The following packages will be UPDATED:

  sphinx                     conda-forge::sphinx-2.4.0-py_0 --> pkgs/main::sphinx-3.0.3-py_0

The following packages will be SUPERSEDED by a higher-priority channel:

  ca-certificates    conda-forge::ca-certificates-2020.4.5~ --> pkgs/main::ca-certificates-2020.1.1-0
  certifi            conda-forge::certifi-2020.4.5.1-py37h~ --> pkgs/main::certifi-2020.4.5.1-py37_0
  openssl            conda-forge::openssl-1.1.1g-h516909a_0 --> pkgs/main::openssl-1.1.1g-h7b6447c_0
  pillow             conda-forge::pillow-7.1.2-py37h718be6~ --> pkgs/main::pillow-7.1.2-py37hb39fc2d_0


Preparing transaction: ...working... done
Verifying transaction: ...working... done
Executing transaction: ...working... done 
```
I don't see any additional options in the [RTD conda docs](https://docs.readthedocs.io/en/stable/guides/conda.html) for overriding this behavior.

**Comment by dotsdl** [comment]
Besides the `.. mdinclude` directive, is there anything else out of `m2r` that we actually need? If not, we could move [the spec](https://github.com/openforcefield/openforcefield/blob/master/The-SMIRNOFF-force-field-format.md) into the doc tree and try sticking with `recommonmark`. This may sidestep the problem entirely.

**Comment by dotsdl** [comment]
Cool, went with one of the workarounds in `conf.py` that appeared to work well enough, and left the other one behind as a comment for now until we have a more permanent solution. As it happens, it appears that API references for methods, but not for e.g. classes, are broken, so we haven't lost all of the internal linking functionality we'd like.

**Pull request merged**